introduce Sub-Project Committer in Contributor Ladder (#320)
cilium/community: CONTRIBUTOR-LADDER.md

    * [Community Contributor](#community-contributor)
    * [Organization Member](#organization-member)
    * [Reviewer](#reviewer)

>     * [Core Scope Repository Committer](#core-scope-repository-committer)

Author: oups

1. Open a PR in the member management file for [the corresponding team](ladder/teams) assigned to a specific review area.
2. At least two members who are already Committers, approve the PR

Automated tooling assigns PRs across the Reviewers defined for each particular area.

### Sub-Project Committer

> Sub-Project Committers have write and merge privileges, and are collectively responsible for steering the repo they are contributing to in a positive direction. As such it represents a significant level of trust in an individual's commitment to working with other committers and the community at large for the benefit of the applicable sub-project.

Author: I picked up the changes from Bill that were very similar (without the explicit tetragon example). Thank you for the suggestion.

Sub-Project Committers are similar to Committers but only within the scope of the repository they contribute to. Note that Sub-Project Committers are not granted the status of Cilium "maintainer" in CNCF parlance. See a list of repositories in [REPOSITORIES.md](REPOSITORIES.md).

Sub-Project Committers have all the rights and responsibilities of a Reviewer, plus:

* **Responsibilities:**

    * Mentoring new Sub-Project Committers
    * Discussing strategy and policy for the sub-project they are contributing to
    * Additional responsibilities specific to any [project roles](CONTRIBUTOR-ROLES.md) they may have
    * Approve contributors to assume roles

* **Qualifications:**

>     * Voted in as laid out in [granting commit access](https://github.com/cilium/community/blob/main/GOVERNANCE.md#granting-commit-access) unless otherwise defined by the sub-project

Reviewer: i was imagining only sub-project committers vote. not sure if cilium committers have enough context to evaluate qualifications of sub-project committer candidates 💭

Automated tooling assigns PRs across the Reviewers defined for each particular area. 

> ### Core Scope Repository Committer

Reviewer 2: ```suggestion
### Sub-Project Committer
```

> Core Scope Repository Committers have write, merge, and voting privileges, and are collectively responsible for steering the repo they are contributing to in a positive direction. As such it represents a significant level of trust in an individual's commitment to working with other committers and the community at large for the benefit of the applicable project.

Reviewer 2: ```suggestion
Sub-Project Committers have write and merge privileges, and are collectively responsible for steering the repo they are contributing to in a positive direction. As such it represents a significant level of trust in an individual's commitment to working with other committers and the community at large for the benefit of the applicable sub-project.
```

> Core Scope Repository Committers are similar to Committers but only within the scope of the Core Scope Repository they contribute to. Note that Core Scope Repository Committers are not granted the status of Cilium "maintainer" in CNCF parlance. Learn mode about the Core Scope in [REPOSITORY-GUIDELINES.md](REPOSITORY-GUIDELINES.md#core) and see a list of Core Scope repositories in [REPOSITORIES.md](REPOSITORIES.md#core-scope).

Reviewer 2: ```suggestion
Sub-Project Committers are similar to Committers but only within the scope of the Core Scope Repository they contribute to. Note that Core Scope Repository Committers are not granted the status of Cilium "maintainer" in CNCF parlance. Learn mode about the Core Scope in [REPOSITORY-GUIDELINES.md](REPOSITORY-GUIDELINES.md#core) and see a list of Core Scope repositories in [REPOSITORIES.md](REPOSITORIES.md#core-scope).
```

Core Scope Repository Committers have all the rights and responsibilities of a Reviewer, plus:

* **Responsibilities:**

    * Mentoring new Core Scope Repository Committers

>     * Discussing strategy and policy for the project they are contributing to

Reviewer 2: ```suggestion
    * Discussing strategy and policy for the sub-project they are contributing to
```

    * [Voting](https://github.com/cilium/community/blob/main/GOVERNANCE.md#voting) on project matters when required
    * Additional responsibilities specific to any [project roles](CONTRIBUTOR-ROLES.md) they may have
    * Approve contributors to assume roles

* **Qualifications:**

>     * Laid out in [granting commit access](https://github.com/cilium/community/blob/main/GOVERNANCE.md#granting-commit-access)

Reviewer 2: I think we might need to write up something different here because right now the process is too tightly tied to become a Committer

Reviewer 3: What do you think about saying that unless the sub-project has more specific instructions, this is the the default instructions?

Then we can separately pursue what we think is a reasonable path for each subproject, and how we document that clearly for future readers of this ladder.
